tests/linux_kernel/helpers/test_cgroup.py

```diff
@@ -112,6 +120,14 @@ def test_cgroup_get_from_path(self):
             NULL(self.prog, "struct cgroup *"),
         )
 
+    def test_cgroup_socket(self):
+        with create_socket() as sock:
+            task = find_task(self.prog, os.getpid())
+            file = fget(task, sock.fileno())
+            sk = cast("struct socket *", file.private_data).sk
+            self.assertTrue(sk_fullsock(sk))
+            self.assertIsNotNone(sock_cgroup_ptr(sk.sk_cgrp_data))
```

Comment:
I think we can do even better and make sure that we get the correct cgroup:

```suggestion
            self.assertEqual(sock_cgroup_ptr(sk.sk_cgrp_data), task.cgroups.dfl_cgrp)
```